libsoftwarecontainer/src/gateway/dbus/dbusgateway.cpp

```diff
@@ -44,12 +55,12 @@ DBusGateway::~DBusGateway()
     }
 }
 
-
 ReturnCode DBusGateway::readConfigElement(const json_t *element)
 {
     DBusGatewayParser parser;
 
-    if (isError(parser.parseDBusConfigElement(element, m_sessionBusConfig, m_systemBusConfig))) {
+    // TODO: This should really be done with exceptions instead and json_t* as return type.
```

Comment:
There's an exception class in gateway.h used by the base class nowadays. Perhaps extend that and use here as you'd like. Or leave it to a separate commit.